### PR TITLE
Refactor URI construction and enhance test coverage

### PR DESCRIPTION
**Pull request by richard-einfinity**
Refactored `DeleteOperation` and `ReplaceOperation` to directly include `operation.ItemId` in URI construction via `MakeAbsoluteUri`. Updated `ExecutableOperation` to support an optional `itemId` parameter in `MakeAbsoluteUri`. Added `MakeAbsoluteUri_WorksWithId` theory test to validate various URI construction scenarios.

### CODE REVIEW

**Comment by richard-einfinity** [comment]
@dotnet-policy-service agree

**Comment by richard-einfinity** [comment]
Resolves #124

**Review by adrianhall — APPROVED** [review]
Will leave this as approved for 24 hours.  If you want to remediate the one comment, it should be a quick fix.

File: src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs [hunk 1]
```diff
@@ -22,19 +22,22 @@ internal abstract class ExecutableOperation
...
+            return new Uri($"{relativeOrAbsoluteUri.ToString().TrimEnd('/')}/{itemId}");
         }
 
         if (baseAddress != null)
         {
             if (baseAddress.IsAbsoluteUri)
             {
-                return new Uri($"{new Uri(baseAddress, relativeOrAbsoluteUri).ToString().TrimEnd('/')}/");
+                return new Uri($"{new Uri(baseAddress, relativeOrAbsoluteUri).ToString().TrimEnd('/')}/{itemId}");
             }
```
[the remark this answers is not in the document]
Comment (richard-einfinity) [reply src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs:41]: Hi @adrianhall,

No problem. I'll update to remove the trailing slash and update the tests. I know in my current environment all the other operations work as expected against the server.

[the remark this answers is not in the document]
Comment (richard-einfinity) [reply src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs:41]: @adrianhall hopefully that's it.

[the remark this answers is not in the document]
Comment (adrianhall) [reply src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs:41]: Nope - this definitely has an issue.  See this one:

`return new Uri($"{relativeOrAbsoluteUri.ToString().TrimEnd('/')}{itemId}");`

If relativeOrAbsoluteUri ends with a / and itemId is set, then there is no separator.

Probably better to construct a string with the right URI in it.  Then `return new Uri(uri.TrimEnd('/'));` at the end.

**Review by adrianhall — CHANGES_REQUESTED** [review]

File: src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs [hunk 2]
```diff
@@ -22,19 +22,22 @@ internal abstract class ExecutableOperation
     /// </summary>
     /// <param name="baseAddress">The base address from the client.</param>
     /// <param name="relativeOrAbsoluteUri">A relative or absolute URI</param>
+    /// <param name="itemId">The id of the item to append to the URI.</param>
     /// <returns></returns>
-    internal static Uri MakeAbsoluteUri(Uri? baseAddress, Uri relativeOrAbsoluteUri)
+    internal static Uri MakeAbsoluteUri(Uri? baseAddress, Uri relativeOrAbsoluteUri, string? itemId = null)
     {
+        itemId = string.IsNullOrWhiteSpace(itemId) ? string.Empty : $"/{itemId}";
```
Comment (richard-einfinity) [inline src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs:29]: @adrianhall the separator is set here if the itemId is specified.

**Comment by adrianhall** [comment]
Explicitly:

```csharp
    internal static Uri MakeAbsoluteUri(Uri? baseAddress, Uri relativeOrAbsoluteUri, string? itemId = null)
    {
        itemId ??= string.Empty;

        string uri = string.Empty;
        if (relativeOrAbsoluteUri.IsAbsoluteUri)
        {
            uri = $"{relativeOrAbsoluteUri.ToString().TrimEnd('/')}/{itemId}";
            return new Uri(uri.TrimEnd('/'));
        }

        if (baseAddress?.IsAbsoluteUri ==true)
        {
              uri = $"{new Uri(baseAddress, relativeOrAbsoluteUri).ToString().TrimEnd('/')}/{itemId}";
              return new Uri(uri.TrimEnd('/'));
        }

        throw new UriFormatException("Invalid combination of baseAddress and relativeUri");
    }
```

File: src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs (hunk 2 above)
Comment (adrianhall) [reply src/CommunityToolkit.Datasync.Client/Offline/Operations/ExecutableOperation.cs:29]: Ah - I get it.  Although, use of IsNullOrWhiteSpace does change the semantics of the method.  If someone submits " ", I believe it gets rejected before this point, so should be ok.

Assuming the tests pass, I'll merge this.

**Review by adrianhall — APPROVED** [review]

**Pull request merged**